OpenRA.Game/Orders/UnitOrderGenerator.cs

```diff
@@ -133,10 +133,11 @@ public virtual bool InputOverridesSelection(World world, int2 xy, MouseInput mi)
 		/// </summary>
 		static UnitOrderResult OrderForUnit(Actor self, Target target, List<Actor> actorsAt, CPos xy, MouseInput mi)
 		{
+			var cmdAlliedUnits = self.World.OrderManager.LobbyInfo.GlobalSettings.OptionOrDefault("cmdalliedunits", false);
```

Comment:
I'm not quite happy with this line. Has anyone a better implementation?

Reply:
Try doing the `cmdAlliedUnits || !a.Owner.IsAlliedWith` inside `TeamTogether` and cache the trait lookup.